### ISSUE TITLE
restrict-plus-operands should apply to template strings

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#3670]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.6.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const foo = {balls: 'balls'};

// Accidentally referenced foo instead of foo.balls
const bar = `bar-${foo}`;
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"restrict-plus-operands": true
	}
}
```

#### Actual behavior

No error is detected, so `bar` gets set to `"bar-[object Object]"` instead of the intended `"bar-balls"`.

#### Expected behavior

Given that the equivalent `const bar = 'bar-' + foo;` would have been flagged by this rule, it seems logical that the example should be too.

If not by default, it would be nice to have this use case covered either by an optional parameter or by a separate rule.

**Comment by OliverJAsh** [comment]
I also just ran into this. Would love to see this rule extended to template strings!

**Comment by buu700** [comment]
Thanks @ColCh (#4807)! This problem was pretty painful.
